ts/store/middlewares/analytics.ts

```diff
@@ -405,7 +407,9 @@ export const actionTracking =
       void trackPaypalOnboarding(mixpanel)(action);
       void trackZendesk(mixpanel)(action);
       void trackCdc(mixpanel)(action);
-      void trackFciAction(mixpanel)(action);
+
+      const fciEnvironment = fciEnvironmentSelector(store.getState());
```

Comment:
Can we use the first parameter to get the state instead of referencing the store?

Reply:
Would you mind explaining this further please?

Reply:
Look at the function signature:
`(_: MiddlewareAPI) => (next: Dispatch) => (action: Action): Action`
if we assign a name to the first function's parameter (e.g., `(middleware: MiddlewareAPI)`), we can do
`const state = middleware.getState()`
and use that instead of referencing and importing the store with `import { store } from "../../boot/configureStoreAndPersistor";`

Reply:
It makes sense, fixed in #025692b, thanks. 
While making this change I also noticed a few more issues: 
- The default `env` value is `prod`, thus each action dispatched before the `getSignatureDetailViewBy` function was tracked with `prod` as environment. I added a default `unknown` value which is used until we obtain the actual environment value. Additionally, it now gets set `unknown` each time a flow finishes or is aborted by the user. This has been fixed in #def1e9e5;
- Some events where being dispatched and tracked twice due to an error in the `put` effect order of the `handleGetSignatureRequestById` saga. This has been addressed in #5dd89da.